### PR TITLE
Bump Mesos to the 1.2.x branch and bump Mesos modules to include ContainerLogger changes.

### PR DESCRIPTION
**Pull request by klueska**
## High Level Description

Bumps dcos-mesos to the Apache Mesos 1.2.x branch to include such critical fixes as:

```
356cf67 Fixed ContainerLogger / IOSwitchboard FD leaks.
9c0bbf1 Removed redundant 'root' user set in containerizer::launch().
0f3a68d Fixed command task with container image 'root' user issue.
801cdfa Added ability to dynamically load '/dev/nvidia-uvm' in GPU isolator.
dc498b7 Ensured the cgroup memory+swap limit is no less than the memory limit.
12890ba Fixed flags logging in Docker executor.
d6b3440 Fixed crash with tasks that use very small resource values.
89d141c Docker environment gets passed on docker run command.
62a8551 Removed containerizer flag logging to prevent leak of sensitive data.
f62f570 Fixed environment overwrite warning to not leak possibly sensitive data.
```

## Related Issues

  - [DCOS-14442](https://jira.mesosphere.com/browse/DCOS-14442) Add support to auto-load /dev/nvidia-uvm and /dev/nvidia-uvm-tools in the GPU isolator.
  - [DCOS-14004](https://jira.mesosphere.com/browse/DCOS-14004) IOSwitchboard FDs leaked when containerizer launch fails -- leads to deadlock.
  - [DCOS_OSS-793](https://jira.mesosphere.com/browse/DCOS_OSS-793) Docker env-file does not allow newlines, but -e did. 
  - [DCOS-14505](https://jira.mesosphere.com/browse/DCOS-14505) UCR mesos-containerizer writes secret values to stdout 
  - [DCOS-14554](https://jira.mesosphere.com/browse/DCOS-14554) Triage why there are so many environment variable overwriting in stdout sandbox.

## Checklist for all PR's

  - [x] Included a test which will fail if code is reverted but test is not. If there is no test please explain here: Relying on Apache Mesos unit tests and existing (and continuously improving) dcos-integration-tests.
  - [x] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [x] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [x] Change log from the last version integrated (this should be a link to commits for easy verification and review):
  * https://github.com/mesosphere/mesos/compare/mesosphere:dcos-mesos-1.2.0...mesosphere:dcos-mesos-1.2.x-c1fd09e
   * https://github.com/dcos/dcos-mesos-modules/compare/36ce01e6...7ce5ba39
  - [x] Test Results: https://jenkins.mesosphere.com/service/jenkins/job/mesos/job/Mesos_CI-build/495/
  - [x] Code Coverage (if available): N/A

### CODE REVIEW

**Comment by klueska** [comment]
Can I get a milestone added to this? @adam-mesos @spahl @lingmann

**Comment by adam-mesos** [comment]
@mesosphere-mergebot bump-ee

**Comment by adam-mesos** [comment]
WIP until CI is green. Looks like you need @greggomann 's SubprocessInfo patch from #1343

**Comment by adam-mesos** [comment]
@mesosphere-mergebot bump-ee

File: packages/mesos-modules/buildinfo.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
     "single_source" : {
       "kind": "git",
       "git": "https://github.com/dcos/dcos-mesos-modules.git",
-      "ref": "36ce01e6cf869b4bb2bfe9e91766f984f333f742",
+      "ref": "5e9d41bc618476686cd9aef318562d5980905517",
       "ref_origin": "master"
```
Comment (adam-mesos) [inline packages/mesos-modules/buildinfo.json]: If the intention is for this to be the bump for DC/OS 1.9, then why aren't you using the `1.9.x` branch of dcos-mesos-modules?
cc: @asridharan @kaysoky

File: packages/mesos/README.md [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
-<li>[09975e642168a47c466e211f02a590b6e3e1e40c] Changed agent_host to expect a relative path.
-<li>[d2cf9ac1dfa0d7d2b5ebcf8f9da86190706bb5cd] Mesos UI: Change paths, pailer, and logs to work with a reverse proxy.
-<li>[3d042bae0431da237d2c109525ea81fe8eb9943c] Revert "Fixed the broken metrics information of master in WebUI."
-<li>[dc01a78680410d7fe49e096508b64e94d991471c] Updated mesos containerizer to ignore GPU isolator creation failure.
+<li>[0b3764964acb8e621bf4af0e66ed29bc6f53d5e7] Set LIBPROCESS_IP into docker container.
+<li>[325ddadc308db79c0a2b57293d94abc3142c6d98] Changed agent_host to expect a relative path.
+<li>[41fd07582ac4357a5a5fe7b86b3ac7f18f179dcb] Mesos UI: Change paths, pailer, and logs to work with a reverse proxy.
+<li>[acb6ac9fbdeb3a97ccfa960d311617d73bb34d53] Revert "Fixed the broken metrics information of master in WebUI."
+<li>[27a6389c996c232acb43a4780f9c5330bca4924b] Updated mesos containerizer to ignore GPU isolator creation failure.
+<li>[6adb3a3071524b87d7ce97d371e82aa2a7af6cbd] Updated the UI to fix maintenance in DC/OS.
```
Comment (adam-mesos) [inline packages/mesos/README.md]: This one-liner will have to be updated again once we bump to a more recent 1.2.x that includes https://github.com/apache/mesos/commit/bec31a51c4c4fc21fa0f08c68d91084f9a559450

File: packages/mesos/buildinfo.json [hunk 3]
```diff
@@ -3,8 +3,8 @@
   "single_source" : {
     "kind": "git",
     "git": "https://github.com/mesosphere/mesos",
-    "ref": "3f504ec18ba5925efbc7245c871cc0583b2f7890",
-    "ref_origin" : "dcos-mesos-1.2.0"
+    "ref": "6adb3a3071524b87d7ce97d371e82aa2a7af6cbd",
+    "ref_origin" : "dcos-mesos-1.2.x-801cdfac"
```
Comment (adam-mesos) [inline packages/mesos/buildinfo.json]: Could be a little closer to today's 1.2.x HEAD, but we'll probably bump it again once @tillt 's env changes land.

File: packages/mesos-modules/buildinfo.json (hunk 1 above)
Comment (asridharan) [inline packages/mesos-modules/buildinfo.json]: Good point @adam-mesos . @klueska could you update the 1.9 tag in the `dcos/dcos-mesos-modules` repo and use the 1.9.x branch (of the 1.9 tag) here?

Comment (kaysoky) [inline packages/mesos-modules/buildinfo.json]: If we're bumping the modules...

Can we include one more commit?
https://github.com/dcos/dcos-mesos-modules/commit/80a69b6ef0afb627f6b0c4cac7dd01cc6a444180

File: packages/mesos/buildinfo.json (hunk 3 above)
Comment (adam-mesos) [inline packages/mesos/buildinfo.json]: Bumped.

File: packages/mesos/README.md (hunk 2 above)
Comment (adam-mesos) [inline packages/mesos/README.md]: Done.

**Comment by adam-mesos** [comment]
@mesosphere-mergebot bump-ee

**Comment by adam-mesos** [comment]
@mesosphere-mergebot bump-ee

**Pull request merged**